SAM-1408

gillmanjc opened the pull request:
I am working on this Jira. I have manually added in the changes provided in the attached patch to a Sakai trunk instance.
The patch however was created from an older sakai-10.x branch, which compared to trunk had some conflicts etc. 
After setting up a test Scenario to test the patch changes, I ran into an issue with authorSettings.jsp not rendering successfully. 
I tried to investigate what the cause might be, because it does not seem like the added changes is the culprit here. I have created an PR.

steveswinsburg commented on samigo/samigo-app/src/webapp/js/extendedTime.js:
```diff
@@ -0,0 +1,208 @@
+		copyListValuesForExtTime(itemNum);
+	}
+
+	var itemNum = 1
```
missing semi colon at end of this line

steveswinsburg commented on samigo/samigo-app/src/webapp/js/extendedTime.js:
```diff
@@ -0,0 +1,208 @@
+	} while(itemNum < allExtendedTimeEntries.length);
+}
+
+// We secretly store the extended time info in a hidden JSF input field. Whenever a user edits an
```
Had to giggle at the 'secretly'. In a file that is sent to the client :+1:

steveswinsburg commented on samigo/samigo-app/src/webapp/js/extendedTime.js:
```diff
@@ -0,0 +1,208 @@
+			"</td></tr></table>"+
+			"</div> <!--end dates -->"+
+			"<hr width=\"450\"  align=\"left\">"+
+			"</div> <!--end xt"+itemNum+" -->";
```
This all seems very fragile :(

steveswinsburg commented:
What is the issue with authorSettings rendering? Its probably all of the javascript it includes. Does the browser JS console give any hints? Maybe step through the JS methods and see whats happening there.

gillmanjc commented:
Hi Steve
I have made some additional changes which seems to be working fine now. Had some page rendering issues with the patch, etc. Please have a look again and let me know.
Will continue with the other Jiras!

steveswinsburg commented:
Hi Jaco,
So in your environment is this all rendering fine now and the feature working well?
cheers,
Steve

gillmanjc commented:
Hi Steve

Everything in the patch provided mentioned in the Jira has been pushed to this branch and is part of this PR. However I am experiencing issues when testing the functionality.

I managed to fix the Calendar popup to be able to choose a date which was not working when I applied the changes from the patch.

Here are my test results:
1) First time link click: "Add a Time Limit/Delivery Date Exception" it only changes to "Add another Time Limit/Delivery Date Exception."
2) Drop down fields in this view has no values for hrs & mins.
3) After selecting date values for "Available Date", "Due Date" & "Retract Date", I choose to save. When I then go back to view the Settings, these date values aren't populated.

I would like to get in contact with the developers who contributed the patch, because it seems like there is still some work to be done to get the Jira resolved other than that of the patch itself. Do you think its best if I copy this comment to the Jira?

Kind Regards
Jaco

steveswinsburg commented:
Hi Jaco,
Yes please put this into the Jira itself. It would probably be good to chat to Paul Dagnall from Dayton University, I believe he worked on this one.

cheers,
Steve

gillmanjc commented:
Hi David

Thank you very much for looking at this Jira and your email reply. 

The issue you mentioned about the error you get in extendedTime.js, was one that I also had. I have changed this value to extendedTimeTarget, and now see changes I didn't see when I tested the patch originally. 

The dropdown lists now gets populated and it values that I select does get saved. I unfortunately haven't had time to test the actual extended hours limitations being configured as part of the new this new patch contribution functionality. 

@Steve, can you please comment on how David can contribute his changes? I think maybe send a diff or patch file to Steve?

steveswinsburg commented:
How big is the change? Can you just attach it here and then it can be incorporated? I don't have the branch locally so I wouldn't be able to integrate it.
Is that fix everything that you need? 

cheers,
Steve

gillmanjc commented:
:) the change was a one line change! (see Updated variable value Id - SAM-1408) I have already committed as part of this PR.

steveswinsburg commented:
Ok I'm ready to pull this down and test it out if you think all code is in
for this one? Let me know and I'll get that done.

On Mon, Aug 31, 2015 at 11:20 PM, Jaco Gillman <notifications@github.com>
wrote:

> :) the change was a one line change! (see Updated variable value Id -
> SAM-1408) I have already committed as part of this PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sakaiproject/sakai/pull/915#issuecomment-136368968>.
>

gillmanjc commented:
Hi Steve, yes please go ahead and pull this down as it is.

steveswinsburg commented:
Hey Jaco, receiving a compilation error:
````
 Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project samigo-app: Compilation failure
[ERROR] /home/travis/build/sakaiproject/sakai/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/util/ExtendedTimeService.java:[9,36] cannot find symbol
[ERROR] symbol:   class AuthzGroupService
[ERROR] location: package org.sakaiproject.authz.cover
````

gillmanjc commented:
Hi Steve
I have switched my workspace branch to SAM-1408 and rebuild a few times, mine does compile just fine. The new class ExtendedTimeService does make use of the deprecated AuthzGroupService but it does however ships with the sakai-kernel-api dependency version 11-SNAPSHOT.

steveswinsburg commented:
Running the build again.

steveswinsburg commented:
It's still failing :(

ncaidin commented:
Fyi, I've assigned the Jira to Jaco G. and set the status to "In Progress" on the ticket.

ern commented:
Need to use rebase instead of merge

steveswinsburg commented:
Ok It's now building However the settings page layout is a bit screwed up. The headings *About*, *Availability*, *Grading* etc are all nested under the top level heading. This is not the case in nightly, they are all separate sections. Could you have a look at the markup?
<img width="332" alt="screen shot 2015-09-28 at 12 24 21 pm" src="https://cloud.githubusercontent.com/assets/129129/10126829/fd01f92a-65db-11e5-91eb-0e6c7c9c733b.png">
<img width="559" alt="screen shot 2015-09-28 at 12 25 07 pm" src="https://cloud.githubusercontent.com/assets/129129/10126830/fd3feb36-65db-11e5-835b-eac9ae5fae69.png">

gillmanjc commented:
Hi Steve

I have committed some more changes after I updated my local workspace with master from trunk. Fixed some issues that weren't in the original patch contribution and certainly not when I tested.

Hopefully this patch contribution now plays along! Please retest and let me know..

ottenhoff commented:
41 commits on this PR :(

Can someone rebase?

steveswinsburg commented:
Closing this. See #1142 for a new PR.
